batchflow/opensets/base.py

```diff
@@ -9,11 +8,14 @@ class Openset(Dataset):
     def __init__(self, index=None, batch_class=None, train_test=False, path=None, preloaded=None, **kwargs):
         self.train_test = train_test
         self._train_index, self._test_index = None, None
-        if preloaded is None:
-            preloaded = self.download(path=path)
-        super().__init__(index, batch_class, preloaded=preloaded, **kwargs)
-        if train_test:
-            self.split()
+
+        if index is None:
+            preloaded, index = self.download(path=path)
+        super().__init__(index, batch_class=batch_class, preloaded=preloaded, **kwargs)
+
+        if self._train_index and self._test_index:
```

Comment:
The fact that you explicitly return `preloaded` and `index` from `download` method, and implicitly set `self._train_index` and `self._test_index` during the same call still makes me uncomfortable

Reply:
That's how things were done before, with only difference having `train_test` flag. 
All the stuff regarding `preloaded`, `index`, `train_index` and `test_index` must be inferred inside `download` method, so potentially it could return 4 values. I found this a bit overwhelming. 
Would like to hear your opinion.

Reply:
This function is ~nevert used be the end user of the `Openset`, so I don't see any problems with it returning as many values as is convenient
A somewhat related example: current `tensorflow.Dataset` classes return tuple of `((train_x, train_y), (test_x, test_y))`